pkg/events/event.go.go

```diff
@@ -0,0 +1,114 @@
+package events
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type Event struct {
+	id           ID // TODO: use id ?
+	id32Bit      ID
+	name         string
+	docPath      string // Relative to the 'doc/events' directory
+	internal     bool
+	syscall      bool
+	dependencies Dependencies
+	sets         []string
+	params       []trace.ArgMeta
+}
+
+func NewEvent(name string, sets []string, depsID []ID) Event {
+	evt := Event{
+		id32Bit: Sys32Undefined,
+		name:    name,
+		sets:    sets,
+	}
+
+	d := Dependencies{
+		Events: make([]ID, 0, len(depsID)),
+	}
+
+	d.Events = append(d.Events, depsID...)
+
+	evt.dependencies = d
+
+	return evt
+}
+
+// NewEvent creates a new Event with all its fields set.
+func NewEventFull(
+	id ID,
+	id32Bit ID,
+	name string,
+	docPath string,
+	internal bool,
+	syscall bool,
+	sets []string,
+	deps Dependencies,
+	params []trace.ArgMeta,
+) Event {
+	// TODO: use id ?
+	return Event{
+		id:           id,
+		id32Bit:      id32Bit,
+		name:         name,
+		docPath:      docPath,
+		internal:     internal,
+		syscall:      syscall,
+		dependencies: deps,
+		sets:         sets,
+		params:       params,
+	}
+}
+
+// Getters (immutable data)
+
+func (e *Event) GetID() ID {
+	return e.id
+}
+
+func (e *Event) GetID32Bit() ID {
+	return e.id32Bit
+}
+
+func (e *Event) GetName() string {
```

Comment:
For getters of private fields, do you think we always need `GetXXX`, could it be `Name()`? wdyt?

Reply:
I like GetXXXX because of IDE completion (I want to get something I type Get and it shows me a list of what I can Get). I think its a good practice and I hope we can make it the default if others accept (on other refactors as well).